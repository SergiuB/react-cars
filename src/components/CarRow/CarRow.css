.car-row {
  cursor: pointer;
}

.car-row:hover {
  background-color: gray;
}