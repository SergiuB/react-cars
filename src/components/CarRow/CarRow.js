import React from 'react';
import PropTypes from 'prop-types';

const CarRow = ({
  name,
  acceleration,
  editEnabled,
  onClick,
}) => (
  <tr className={`car-row ${editEnabled ? ' edit-enabled' : ''}`} onClick={onClick}>
    <td className='text-col'>{name}</td>
    <td className='numeric-col'>{acceleration}</td>
  </tr>
);

CarRow.propTypes = {
  name: PropTypes.string.isRequired,
  acceleration: PropTypes.number.isRequired,
  editEnabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

CarRow.defaultProps = {
  editEnabled: false,
  onClick: () => {}
};

export default CarRow;
