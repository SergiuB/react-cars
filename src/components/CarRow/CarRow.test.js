import React from 'react';
import { shallow } from 'enzyme';

import CarRow from './CarRow';

describe('CarRow', function () {
  it('renders correctly', function () {
    const wrapper = shallow(
      <CarRow
        id={1}
        name='Audi'
        acceleration={12}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly in edit mode', function () {
    const wrapper = shallow(
      <CarRow
        id={1}
        name='Audi'
        acceleration={12}
        editEnabled
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('fires onClick when row is clicked', function () {
    const onClick = jest.fn();
    const wrapper = shallow(
      <CarRow
        id={1}
        name='Audi'
        acceleration={12}
        onClick={onClick}
      />
    );
    wrapper.find('tr').simulate('click');
    expect(onClick).toBeCalled();
  });
});
