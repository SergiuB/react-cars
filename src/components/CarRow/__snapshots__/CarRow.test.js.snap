// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarRow renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <tr
    className="car-row "
    onClick={[Function]}
>
    <td
        className="text-col"
    >
        Audi
    </td>
    <td
        className="numeric-col"
    >
        12
    </td>
</tr>,
  "nodes": Array [
    <tr
      className="car-row "
      onClick={[Function]}
>
      <td
            className="text-col"
      >
            Audi
      </td>
      <td
            className="numeric-col"
      >
            12
      </td>
</tr>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CarRow
        acceleration={12}
        editEnabled={false}
        name="Audi"
        onClick={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "acceleration": 12,
          "editEnabled": false,
          "name": "Audi",
          "onClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <tr
          className="car-row "
          onClick={[Function]}
>
          <td
                    className="text-col"
          >
                    Audi
          </td>
          <td
                    className="numeric-col"
          >
                    12
          </td>
</tr>,
        "_debugID": 2,
        "_renderedOutput": <tr
          className="car-row "
          onClick={[Function]}
>
          <td
                    className="text-col"
          >
                    Audi
          </td>
          <td
                    className="numeric-col"
          >
                    12
          </td>
</tr>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CarRow
    acceleration={12}
    editEnabled={false}
    name="Audi"
    onClick={[Function]}
/>,
}
`;

exports[`CarRow renders correctly in edit mode 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <tr
    className="car-row  edit-enabled"
    onClick={[Function]}
>
    <td
        className="text-col"
    >
        Audi
    </td>
    <td
        className="numeric-col"
    >
        12
    </td>
</tr>,
  "nodes": Array [
    <tr
      className="car-row  edit-enabled"
      onClick={[Function]}
>
      <td
            className="text-col"
      >
            Audi
      </td>
      <td
            className="numeric-col"
      >
            12
      </td>
</tr>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CarRow
        acceleration={12}
        editEnabled={true}
        name="Audi"
        onClick={[Function]}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "acceleration": 12,
          "editEnabled": true,
          "name": "Audi",
          "onClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <tr
          className="car-row  edit-enabled"
          onClick={[Function]}
>
          <td
                    className="text-col"
          >
                    Audi
          </td>
          <td
                    className="numeric-col"
          >
                    12
          </td>
</tr>,
        "_debugID": 4,
        "_renderedOutput": <tr
          className="car-row  edit-enabled"
          onClick={[Function]}
>
          <td
                    className="text-col"
          >
                    Audi
          </td>
          <td
                    className="numeric-col"
          >
                    12
          </td>
</tr>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CarRow
    acceleration={12}
    editEnabled={true}
    name="Audi"
    onClick={[Function]}
/>,
}
`;
